src/librustdoc/clean/utils.rs

```diff
@@ -411,7 +411,11 @@ crate fn resolve_type(cx: &mut DocContext<'_>, path: Path, id: hir::HirId) -> Ty
             return Generic(kw::SelfUpper);
         }
         Res::Def(DefKind::TyParam, _) if path.segments.len() == 1 => {
-            return Generic(Symbol::intern(&format!("{:#}", path.print(&cx.cache, cx.tcx))));
+            let relative_to = vec![];
```

Comment:
Hmm, what happens if you call resolve_type() from a submodule? Or is that not possible because cleaning happens before rendering?

Reply:
I don't think this matters. It's using the alternate printing style (the one with the `#`), and that doesn't call `href()`.

I'll add a comment to point this out.